internal/cf/resourcecache_test.go

```diff
@@ -0,0 +1,210 @@
+/*
+SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and cf-service-operator contributors
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cf
+
+import (
+	"strconv"
+	"sync"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/sap/cf-service-operator/internal/facade"
+)
+
+func TestFacade(t *testing.T) {
+	RegisterFailHandler(Fail)
+	//RunSpecs(t, "ResourceCache Suite")
+}
+
+var _ = Describe("ResourceCache", func() {
+	var cache *resourceCache
+	var instance *facade.Instance
+	var wg sync.WaitGroup
+	concurrencyLevel := 20
+
+	BeforeEach(func() {
+		cache = initResourceCache()
+		instance = &facade.Instance{
+			Guid:            "guid1",
+			Name:            "name1",
+			Owner:           "owner1",
+			ServicePlanGuid: "plan1",
+			ParameterHash:   "hash1",
+			Generation:      1,
+		}
+	})
+
+	Context("basic CRUD operations", func() {
+		It("should add, get, update, and delete an instance in the cache", func() {
+			// Add instance
+			Ownerkey := "owner1"
+			cache.addInstanceInCache(Ownerkey, instance)
+
+			// Get instance
+			retrievedInstance, found := cache.getInstanceFromCache(Ownerkey)
+			Expect(found).To(BeTrue())
+			Expect(retrievedInstance).To(Equal(instance))
+
+			// Update instance
+			updatedInstance := &facade.Instance{
+				Guid:            "guid1",
+				Name:            "updatedName",
+				Owner:           "owner1",
+				ServicePlanGuid: "updatedPlan",
+				ParameterHash:   "hash1",
+				Generation:      2,
+			}
+			cache.updateInstanceInCache("guid1", "updatedName", "owner1", "updatedPlan", nil, 2)
+			retrievedInstance, found = cache.getInstanceFromCache(Ownerkey)
+			Expect(found).To(BeTrue())
+			Expect(retrievedInstance).To(Equal(updatedInstance))
+
+			// Delete instance
+			cache.deleteInstanceFromCache(Ownerkey)
+			_, found = cache.getInstanceFromCache(Ownerkey)
+			Expect(found).To(BeFalse())
+		})
+	})
+
+	Context("edge cases", func() {
+		It("should handle adding an instance with an existing key", func() {
+			cache.addInstanceInCache("owner1", instance)
+			cache.addInstanceInCache("owner1", instance)
```

Comment:
You could add two different instance objects using the same key and expect that the last one wins.

Reply:
Added the expect in commit https://github.com/SAP/cf-service-operator/pull/59/commits/f86cf7671fcf8c1cb15d1c42a65165bc4d5991d4